File: mobile_android/lib-maplat/src/main/java/jp/tilemap/maplat/MaplatView.java
package jp.tilemap.maplat;

import android.content.Context;
import android.os.Handler;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.webkit.WebView;
import android.widget.RelativeLayout;

import java.util.ArrayList;
import java.util.HashMap;

/**
 * MaplatのWebViewを内包したView
 *
 * @author Ishimaru Sohei
 */
public class MaplatView extends RelativeLayout {

    private MaplatBridge mMaplatBridge;

    /** 現在のマップの名称 */
    private String mNowMap;
    /** WebView */
    private WebView mWebView;

    public MaplatView(@NonNull Context context) {
        this(context, null, 0);
    }

    public MaplatView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public MaplatView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    public void setMaplatBridgeListener(MaplatBridgeListener maplatBridgeListener) {
        mMaplatBridge.setMaplatBridgeListener(maplatBridgeListener);
    }

    public MaplatBridge getMaplatBridge() {
        return mMaplatBridge;
    }

    public void setMaplatBridge(MaplatBridge maplatBridge) {
        this.mMaplatBridge = maplatBridge;
    }

    /**
     * MaplatBridgeの初期化
     */
    public void initSetting(MaplatBridgeListener listner, String appID, HashMap<String, Object> setting) {
        try {
            mMaplatBridge = new MaplatBridge(getContext(), mWebView, new Handler(), listner, appID, setting);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    /**
     * WebViewの初期化
     */
    private void init() {
        mWebView = new WebView(getContext());
        WebView.setWebContentsDebuggingEnabled(true);
        addView(mWebView, new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
    }

    public void changeMap(String mapID) {
        mMaplatBridge.changeMap(mapID);
    }

    public void addMarker(double latitude, double longitude, long markerId, long markerData) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData);
    }
    public void addMarker(double latitude, double longitude, long markerId, long markerData, String iconUrl) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl);
    }
    public void addMarker(double latitude, double longitude, long markerId, double markerData) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData);
    }
    public void addMarker(double latitude, double longitude, long markerId, double markerData, String iconUrl) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl);
    }
    public void addMarker(double latitude, double longitude, long markerId, String markerData) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData);
    }
    public void addMarker(double latitude, double longitude, long markerId, String markerData, String iconUrl) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl);
    }
    public void addMarker(double latitude, double longitude, long markerId, ArrayList<Object> markerData) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData);
    }
    public void addMarker(double latitude, double longitude, long markerId, ArrayList<Object> markerData, String iconUrl) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl);
    }
    public void addMarker(double latitude, double longitude, long markerId, HashMap<String, Object> markerData) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData);
    }
    public void addMarker(double latitude, double longitude, long markerId, HashMap<String, Object> markerData, String iconUrl) {
        mMaplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl);
    }

    public void clearMarker() {
        mMaplatBridge.clearMarker();
    }
    public void clearLine() {
        mMaplatBridge.clearLine();
    }

    public void setViewpoint(double latitude, double longitude) {
        mMaplatBridge.setViewpoint(latitude, longitude);
    }
    public void setDirection(double direction) {
        mMaplatBridge.setDirection(direction);
    }
    public void setRotation(double rotation) {
        mMaplatBridge.setRotation(rotation);
    }

    public void setGPSMarker(double latitude, double longitude, double accuracy) {
        mMaplatBridge.setGPSMarker(latitude, longitude, accuracy);
    }
    public void addLine(ArrayList<ArrayList<Double>> lnglats, HashMap<String, Object> stroke) {
        mMaplatBridge.addLine(lnglats, stroke);
    }

    public void currentMapInfo(IMaplatMapCallbackHandler callback) {
        mMaplatBridge.currentMapInfo(callback);
    }

    public void mapInfo(String sourceID, IMaplatMapCallbackHandler callback) {
        mMaplatBridge.mapInfo(sourceID, callback);
    }
}

File: mobile_android/lib-maplat/src/main/java/jp/tilemap/maplat/MaplatView.kt
package jp.tilemap.maplat

import android.content.Context
import android.os.Handler
import android.util.AttributeSet
import android.webkit.WebView
import android.widget.RelativeLayout

import java.util.ArrayList
import java.util.HashMap

/**
 * MaplatのWebViewを内包したView
 *
 * @author Ishimaru Sohei
 */
class MaplatView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : RelativeLayout(context, attrs, defStyleAttr) {

    lateinit var maplatBridge: MaplatBridge

    /** 現在のマップの名称  */
    private val mNowMap: String? = null
    /** WebView  */
    lateinit var mWebView: WebView

    init {
        init()
    }

    fun setMaplatBridgeListener(maplatBridgeListener: MaplatBridgeListener) {
        maplatBridge.setMaplatBridgeListener(maplatBridgeListener)
    }

    /**
     * MaplatBridgeの初期化
     */
    fun initSetting(listner: MaplatBridgeListener, appID: String, setting: HashMap<String, Any>) {
        try {
            maplatBridge = MaplatBridge(context, mWebView, Handler(), listner, appID, setting)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    /**
     * WebViewの初期化
     */
    private fun init() {
        mWebView = WebView(context)
        WebView.setWebContentsDebuggingEnabled(true)
        addView(mWebView, LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT))
    }

    fun changeMap(mapID: String) {
        maplatBridge.changeMap(mapID)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: Long) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: Long, iconUrl: String) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: Double) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: Double, iconUrl: String) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: String) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: String, iconUrl: String) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: ArrayList<Any>) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: ArrayList<Any>, iconUrl: String) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: HashMap<String, Any>) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData)
    }

    fun addMarker(latitude: Double, longitude: Double, markerId: Long, markerData: HashMap<String, Any>, iconUrl: String) {
        maplatBridge.addMarker(latitude, longitude, markerId, markerData, iconUrl)
    }

    fun clearMarker() {
        maplatBridge.clearMarker()
    }

    fun clearLine() {
        maplatBridge.clearLine()
    }

    fun setViewpoint(latitude: Double, longitude: Double) {
        maplatBridge.setViewpoint(latitude, longitude)
    }

    fun setDirection(direction: Double) {
        maplatBridge.setDirection(direction)
    }

    fun setRotation(rotation: Double) {
        maplatBridge.setRotation(rotation)
    }

    fun setGPSMarker(latitude: Double, longitude: Double, accuracy: Double) {
        maplatBridge.setGPSMarker(latitude, longitude, accuracy)
    }

    fun addLine(lnglats: ArrayList<ArrayList<Double>>, stroke: HashMap<String, Any>?) {
        maplatBridge.addLine(lnglats, stroke)
    }

    fun currentMapInfo(callback: IMaplatMapCallbackHandler) {
        maplatBridge.currentMapInfo(callback)
    }

    fun mapInfo(sourceID: String, callback: IMaplatMapCallbackHandler) {
        maplatBridge.mapInfo(sourceID, callback)
    }
}
